Automattic/woocommerce-payments
# Add shopper Tracks events

Fixes https://github.com/Automattic/woopay/issues/1907

#### Changes proposed in this Pull Request

This PR adds the following Tracks events
```
wcpay_cart_page_view
wcpay_product_page_view
wcpay_gpay_button_click
wcpay_apple_pay_button_click
wcpay_checkout_email_address_woopay_check
wcpay_checkout_woopay_save_my_info_tooltip_hover
wcpay_checkout_woopay_save_my_info_tooltip_learn_more_click
wcpay_checkout_woopay_save_my_info_privacy_policy_click
wcpay_checkout_save_my_info_tos_click
```








#### Testing instructions





Save My Info events

* Add a product to the cart
* Visit shortcode checkout page as an unregistered WooPay user.
* Hover over the "i" tooltip in Save My Info section.
* Click the Learn More link in the tooltip. (`wcpay_checkout_woopay_save_my_info_tooltip_hover`)
* Check the save my info checkbox. 
* Click on Terms of Service and Privacy Policy links.  (`wcpay_checkout_save_my_info_tos_click`, `wcpay_checkout_woopay_save_my_info_privacy_policy_click`)
* In ~5 minutes, check MC Tracks live view and make sure above events are recorded.
* Repeat the tests for Blocks checkout.

Payment Request Buttons

* Click on the Google Pay button from product, cart and checkout page.
* Click on the Apple Pay button from the product, cart and checkout page.
* Check Tracks Live view for `wcpay_gpay_button_click` and `wcpay_apple_pay_button_click` with the relevant `source` property indicating the page.

Page Views
* View the cart page.
* View a product page.
* Check MC Live view for `wcpay_cart_page_view` and `wcpay_product_page_view` events.

-------------------


**Post merge**
